docs/developer/watcher-local-test.md

```diff
@@ -1,142 +1,167 @@
-```
-2. Run the following command to create a local kyma-runtime (SKR) cluster:
-```shell
-k3d cluster create skr-local
-```
-### Switch context to the KCP cluster config
-Run the following command to switch `kubectl` config context to prepare for running the subsequent commands:
-```shell
-kubectl config use-context k3d-kcp-local
-```
-### Install Istio and Manifest CRD
-1. Run the following command to install `istioctl` using `brew`:
-```shell
-brew install istioctl
-```
-2. Run the following command to install Istio and Manifest CRD on the KCP cluster:
-```shell
-istioctl install --set profile=demo -y && \
-kubectl apply -f https://raw.githubusercontent.com/kyma-project/module-manager/main/config/crd/bases/operator.kyma-project.io_manifests.yaml
-```
+### KCP cluster setup
+   
+1. Create a local control-plane (KCP) cluster:
+    ```shell
+    k3d cluster create kcp-local --port 9080:80@loadbalancer \
+    --registry-create k3d-registry.localhost:0.0.0.0:5111 \
+    --k3s-arg '--no-deploy=traefik@server:0'
+    ```
+   
+2. Open `/etc/hosts` file on your local system:
+    ```shell
+    sudo nano /etc/hosts
+   ```
+   Add entry for your local k3d registry created in the previous steps
+   ```
+    127.0.0.1       k3d-registry.localhost
+   ```
 
-### Deploy lifecycle-manager on KCP:
-Run the following command to deploy lifecycle-manager on the KCP cluster:
-```shell
-make local-deploy-with-watcher IMG=eu.gcr.io/kyma-project/lifecycle-manager:latest
-```
+3. Install other pre-requisites required by the lifecycle-manager
+   1. `Istio` CRDs using `istioctl`
+      ```shell
+      brew install istioctl && \
+      istioctl install --set profile=demo -y
+      ```
+   2. `module-manager` CRDs
+      ```shell
+       kubectl apply -f https://raw.githubusercontent.com/kyma-project/module-manager/main/config/crd/bases/operator.kyma-project.io_manifests.yaml
+      ```
+      
+4. Deploy lifecycle-manager on the cluster:
+    ```shell
+    make local-deploy-with-watcher IMG=eu.gcr.io/kyma-project/lifecycle-manager:latest
+    ```
 
-### Apply sample module templates for sample-kyma:
-Run the following commands to apply sample module template needed for sample Kyma on the KCP cluster:
-```shell
-kubectl apply -f https://raw.githubusercontent.com/kyma-project/lifecycle-manager/main/config/samples/component-integration-installed/operator_v1alpha1_moduletemplate_kcp-module.yaml
-```
+5. Create `module-template` by using [kyma-cli](https://github.com/kyma-project/cli)
+   which serves the role of a component-descriptor for module installations.
+
+   For this setup, we will create a module template from the [template-operator](https://github.com/kyma-project/template-operator) repository as reference.
+   Adjust your path to your template-operator local directory or any other reference module operator accordingly.
+
+    ```shell
+   kyma alpha create module -p ../template-operator --version 1.2.3 -w --registry k3d-registry.localhost:5111 --insecure
```

Comment:
The latest kyma-cli issues the following prompt here:
```
WARNING: This command is experimental and might change in its final version. Use at your own risk.
- Kustomize ready
? Please specify the file to use as default CR in /[edited]/golang/src/github.com/kyma-project/template-operator/config/samples: [operator.kyma-project.io_v1alpha1_sample.yaml operator.kyma-project.io_v1alpha1_samplehelm.yaml]
```
Which CR should be used?

Reply:
You can use any CR. I think the choice comes with the latest merge in https://github.com/kyma-project/template-operator/pull/14